wpilibc/athena/src/SPI.cpp

```diff
@@ -147,7 +148,16 @@ int SPI::Write(uint8_t* data, int size) {
 int SPI::Read(bool initiate, uint8_t* dataReceived, int size) {
   int retVal = 0;
   if (initiate) {
-    auto dataToSend = new uint8_t[size];
+    // Create a stack allocation if the data to transmit is small.
+    uint8_t stackData[32];
+    std::unique_ptr<uint8_t[]> heapData;
+    uint8_t* dataToSend;
+    if (size > 32) {
+      heapData = std::make_unique<uint8_t[]>(size);
+      dataToSend = heapData.get();
+    } else {
+      dataToSend = stackData;
+    }
     std::memset(dataToSend, 0, size);
```

Comment:
This is the case were the user has requested a read, but also needs an initialization of 0's to happen. This is the proper behavior.